cli/tsc/99_main_compiler.js

```diff
@@ -471,6 +471,10 @@ delete Object.prototype.__proto__;
     core.jsonOpSync("op_respond", { id, data });
   }
 
+  const userPreferences = {
```

Comment:
```suggestion
  /** @type {ts.UserPreferences} */
  const userPreferences = {
```

I think it is also debatable if this were moved to Rust as well, as sort of want zero config in JavaScript if we can avoid it.

Reply:
As we will likely need to change/update/manage these on the lsp side, especially when we need to do refactorings... We could do it like the configuration managements as well...

```ts
export interface UserPreferences {
        readonly disableSuggestions?: boolean;
        readonly quotePreference?: "auto" | "double" | "single";
        readonly includeCompletionsForModuleExports?: boolean;
        readonly includeAutomaticOptionalChainCompletions?: boolean;
        readonly includeCompletionsWithInsertText?: boolean;
        readonly importModuleSpecifierPreference?: "auto" | "relative" | "non-relative";
        /** Determines whether we import `foo/index.ts` as "foo", "foo/index", or "foo/index.js" */
        readonly importModuleSpecifierEnding?: "auto" | "minimal" | "index" | "js";
        readonly allowTextChangesInNewFiles?: boolean;
        readonly providePrefixAndSuffixTextForRename?: boolean;
        readonly includePackageJsonAutoImports?: "auto" | "on" | "off";
        readonly provideRefactorNotApplicableReason?: boolean;
    }
```

Reply:
Good point. Let me move it.